CPAC/nuisance/utils/compcor.py

```diff
@@ -181,9 +181,34 @@ def fallback_svd(a, full_matrices=True, compute_uv=True):
 
 
 def TR_string_to_float(tr):
-    if 'ms' in tr:
-        tr = float(tr.replace('ms', '')) / 1000
-    else:
-        tr = float(tr.replace('s', ''))
+    """
+    Convert TR string to seconds (float). Suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Parameters
+    ----------
+    tr : TR string representation. May use suffixes 's' or 'ms' to indicate
+    seconds or milliseconds.
+
+    Returns
+    -------
+    tr in seconds (float)
+    """
```

Comment:
While we're adding a docstring, this is a good opportunity to throw in some doctests too, to unit test this function

- [ ] with 's'
- [ ] with 'ms'
- [ ] with one or more types that is not string

Reply:
Added tests with 2c7d373